aws-xray-recorder-sdk-core/src/test/java/com/amazonaws/xray/AWSXRayRecorderTest.java

```diff
@@ -443,83 +447,83 @@ public void testEmittingSegmentTwiceThrowsSegmentAlreadyEmittedException() {
     }
 
     @Test
-    public void testSubsegmentFunctionExceptionWhenMissingContextIsLogged() {
-        // given
-        RuntimeException expectedException = new RuntimeException("To be thrown by function");
+    public void testBeginSegmentWhenMissingContext() {
+        AWSXRayRecorder recorder = AWSXRayRecorderBuilder.standard()
+                                                         .withSegmentContextResolverChain(new SegmentContextResolverChain())
+                                                         .withContextMissingStrategy(new IgnoreErrorContextMissingStrategy())
+                                                         .build();
+        Segment segment = recorder.beginSegment("hello");
+        assertThat(segment).isNotNull();
+        assertThat(segment.getNamespace()).isEmpty();
+        // No-op
+        segment.setNamespace("foo");
+        assertThat(segment.getNamespace()).isEmpty();
+    }
+
+    @Test
+    public void testBeginSubsegmentWhenMissingContext() {
+        AWSXRayRecorder recorder = AWSXRayRecorderBuilder.standard()
+                                                         .withContextMissingStrategy(new IgnoreErrorContextMissingStrategy())
+                                                         .build();
+        Subsegment subsegment = recorder.beginSubsegment("hello");
+        assertThat(subsegment).isNotNull();
+        assertThat(subsegment.getNamespace()).isEmpty();
+        // No-op
+        subsegment.setNamespace("foo");
+        assertThat(subsegment.getNamespace()).isEmpty();
+    }
+
+    @Test
+    public void testSubsegmentFunctionExceptionWhenMissingContext() {
+        IllegalStateException expectedException = new IllegalStateException("To be thrown by function");
         Function<Subsegment, Void> function = (subsegment) -> {
             throw expectedException;
         };
+
         AWSXRayRecorder recorder = AWSXRayRecorderBuilder.standard()
-                                                         .withContextMissingStrategy(new LogErrorContextMissingStrategy())
+                                                         .withContextMissingStrategy(new IgnoreErrorContextMissingStrategy())
```

Comment:
These tests don't actually test whether anything's logged so I just removed that to make them more precise.